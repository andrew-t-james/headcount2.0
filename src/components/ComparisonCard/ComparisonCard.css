.comparison-card-message {
  font-size: 3rem;
  padding: 0 4rem;
}

@media screen and (max-width: 560px) {
  .comparison-card-message {
    font-size: 2rem;
    padding-top: 5rem;
  }
}