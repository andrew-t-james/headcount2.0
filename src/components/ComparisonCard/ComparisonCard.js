import React from 'react';
import PropTypes from 'prop-types';

import './ComparisonCard.css';

const ComparisonCard = props => {
  const comparedDistrictKeys = Object.keys(props);
  if (!comparedDistrictKeys.length) {
    return (
      <div className="comparison-card">
        <h2 className="comparison-card-message">Please Add a District to Compare</h2>
      </div>
    );
  }
  return (

    <div>
      { comparedDistrictKeys[0] &&
        <h2>{comparedDistrictKeys[0]} : {props[comparedDistrictKeys[0]]}</h2>
      }
      { comparedDistrictKeys[2] &&
        <h2>Comparison : {props[comparedDistrictKeys[2]]}</h2>
      }
      { comparedDistrictKeys[1] &&
        <h2>{comparedDistrictKeys[1]} : {props[comparedDistrictKeys[1]]}</h2>
      }
    </div>
  );
};

ComparisonCard.propTypes = {
  props: PropTypes.object
};

export default ComparisonCard;
