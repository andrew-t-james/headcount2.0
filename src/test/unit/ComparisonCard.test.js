import React from 'react';
import { shallow } from 'enzyme';
import ComparisonCard from '../../components/ComparisonCard/ComparisonCard';

describe('GraphCard unit test suite', () => {
  let wrapper;
  const mockDistrict = {
    'ADAMS 20': 0.294,
    'COLORADO': 0.234,
    compared: 1.345
  };

  beforeEach(() => wrapper = shallow(<ComparisonCard {...mockDistrict}/>));

  afterEach(() => wrapper.unmount());

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should match snapshot when no data is passed', () => {
    wrapper = shallow(<ComparisonCard {...{}} />);
    expect(wrapper).toMatchSnapshot();
  });

});
