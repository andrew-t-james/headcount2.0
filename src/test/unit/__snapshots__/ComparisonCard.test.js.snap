// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphCard unit test suite should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    ADAMS 20={0.294}
    COLORADO={0.234}
    compared={1.345}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          ADAMS 20
           : 
          0.294
</h2>,
        <h2>
          Comparison : 
          1.345
</h2>,
        <h2>
          COLORADO
           : 
          0.234
</h2>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ADAMS 20",
            " : ",
            0.294,
          ],
        },
        "ref": null,
        "rendered": Array [
          "ADAMS 20",
          " : ",
          0.294,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Comparison : ",
            1.345,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Comparison : ",
          1.345,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "COLORADO",
            " : ",
            0.234,
          ],
        },
        "ref": null,
        "rendered": Array [
          "COLORADO",
          " : ",
          0.234,
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            ADAMS 20
             : 
            0.294
</h2>,
          <h2>
            Comparison : 
            1.345
</h2>,
          <h2>
            COLORADO
             : 
            0.234
</h2>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ADAMS 20",
              " : ",
              0.294,
            ],
          },
          "ref": null,
          "rendered": Array [
            "ADAMS 20",
            " : ",
            0.294,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Comparison : ",
              1.345,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Comparison : ",
            1.345,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "COLORADO",
              " : ",
              0.234,
            ],
          },
          "ref": null,
          "rendered": Array [
            "COLORADO",
            " : ",
            0.234,
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`GraphCard unit test suite should match snapshot when no data is passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2
        className="comparison-card-message"
>
        Please Add a District to Compare
</h2>,
      "className": "comparison-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Please Add a District to Compare",
        "className": "comparison-card-message",
      },
      "ref": null,
      "rendered": "Please Add a District to Compare",
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2
          className="comparison-card-message"
>
          Please Add a District to Compare
</h2>,
        "className": "comparison-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please Add a District to Compare",
          "className": "comparison-card-message",
        },
        "ref": null,
        "rendered": "Please Add a District to Compare",
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
